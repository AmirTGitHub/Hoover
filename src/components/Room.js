import React from 'react';
import Cell from './Cell'


const Room = (props) => {
    return (
        <div style={{ width: `${props.width ? props.width * 51 : 100}px` }} className='roomFloor' >
            {props.grid ? props.grid.map((item, index) => {
                return (

                    <Cell key={index} position={index + 1} hoover={props.hoover} dirt={props.dirt} />

                )
            }) : null}
        </div>
    )
}

export default Room