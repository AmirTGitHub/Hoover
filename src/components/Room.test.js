import Room from './Room'


const roomSetup = {
    grid: [...new Array(25)],
    width: 5
}

it('render correctly the room', () => {
    const wrapper = render(<Room {...roomSetup} />)
    expect(wrapper).toMatchSnapshot()

})

it('not render anything if there was any props', () => {
    const wrapper = render(<Room />)
    expect(wrapper).toMatchSnapshot()
})

it('render anything if grid is 0', () => {
    const wrapper = render(<Room grid={0} />)
    expect(wrapper).toMatchSnapshot()
})