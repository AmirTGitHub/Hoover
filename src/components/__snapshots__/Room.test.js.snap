// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not render anything if there was any props 1`] = `
<div
  class="roomFloor"
  style="width:100px"
>
  
</div>
`;

exports[`render anything if grid is 0 1`] = `
<div
  class="roomFloor"
  style="width:100px"
>
  
</div>
`;

exports[`render correctly the room 1`] = `
<div
  class="roomFloor"
  style="width:255px"
>
  <div
    class="cell"
    id="1"
  >
    
  </div>
  <div
    class="cell"
    id="2"
  >
    
  </div>
  <div
    class="cell"
    id="3"
  >
    
  </div>
  <div
    class="cell"
    id="4"
  >
    
  </div>
  <div
    class="cell"
    id="5"
  >
    
  </div>
  <div
    class="cell"
    id="6"
  >
    
  </div>
  <div
    class="cell"
    id="7"
  >
    
  </div>
  <div
    class="cell"
    id="8"
  >
    
  </div>
  <div
    class="cell"
    id="9"
  >
    
  </div>
  <div
    class="cell"
    id="10"
  >
    
  </div>
  <div
    class="cell"
    id="11"
  >
    
  </div>
  <div
    class="cell"
    id="12"
  >
    
  </div>
  <div
    class="cell"
    id="13"
  >
    
  </div>
  <div
    class="cell"
    id="14"
  >
    
  </div>
  <div
    class="cell"
    id="15"
  >
    
  </div>
  <div
    class="cell"
    id="16"
  >
    
  </div>
  <div
    class="cell"
    id="17"
  >
    
  </div>
  <div
    class="cell"
    id="18"
  >
    
  </div>
  <div
    class="cell"
    id="19"
  >
    
  </div>
  <div
    class="cell"
    id="20"
  >
    
  </div>
  <div
    class="cell"
    id="21"
  >
    
  </div>
  <div
    class="cell"
    id="22"
  >
    
  </div>
  <div
    class="cell"
    id="23"
  >
    
  </div>
  <div
    class="cell"
    id="24"
  >
    
  </div>
  <div
    class="cell"
    id="25"
  >
    
  </div>
</div>
`;
